src/irr/asset/COBJMeshFileLoader.cpp

```diff
@@ -39,13 +39,361 @@ namespace std
 }
 */
 
-#define NEW_SHADERS
+namespace
+{
+    constexpr const char* VERT_SHADER_NO_UV = 
+R"(#version 430 core
+
+layout (location = 0) in vec3 vPos;
+layout (location = 3) in vec3 vNormal;
+
+layout (location = 0) out vec3 LocalPos;
+layout (location = 1) out vec3 ViewPos;
+layout (location = 2) out vec3 Normal;
+
+layout (set = 1, binding = 0, row_major, std140) uniform UBO {
+    mat4 MVP;
+    mat4 MV;
+    mat3 NormalMat;
+    vec3 EyePos;
+} CamData;
+
+void main()
+{
+    LocalPos = vPos;
+    gl_Position = CamData.MVP*vec4(vPos, 1.0);
+    ViewPos = (CamData.MV*vec4(vPos, 1.0)).xyz;
```

Comment:
`matrix*vec4(v,1.0)` needs its own function and header (`vertex_utils.glsl` or something)

Reply:
the proper way to factor it out is
`matrix[0]*v.x+matrix[1]*v.y+matrix[2]*v.z+matrix[3]`

Reply:
furthermore the `CamData` built-in header should have a function that does the MVP and MV transformations on some input vector (and uses the `vertex_utils.glsl`)